File: packages/rune-async-storage/ios/RuneAsyncStorage.swift
//
//  RuneAsyncStorage.swift
//  RuneAsyncStorage
//
//  Public interface for RuneAsyncStorage module
//

import Foundation
import RuneKit

@objc(RuneAsyncStorage)
public class RuneAsyncStorage: NSObject {
  private static var moduleInstance: RuneAsyncStorageModule?

  @objc public static func initialize(with runtime: RuneRuntime) {
    guard moduleInstance == nil else {
      print("[RuneAsyncStorage] Module already initialized")
      return
    }

    let module = RuneAsyncStorageModule()
    runtime.installModules([module])
    moduleInstance = module
    print("[RuneAsyncStorage] Module initialized")
  }

  @objc public static func cleanup() {
    moduleInstance = nil
  }
}

File: packages/rune-async-storage/android/RuneAsyncStorage/src/main/java/dev/rune/asyncstorage/RuneAsyncStorage.kt
package dev.rune.asyncstorage

import android.app.Activity
import com.rune.kit.runtime.RuneRuntime

/**
 * Public interface for RuneAsyncStorage module.
 */
object RuneAsyncStorage {
    private var moduleInstance: RuneAsyncStorageModule? = null

    /**
     * Initialize the AsyncStorage module with an Activity and RuneRuntime instance.
     */
    @JvmStatic
    fun initialize(activity: Activity, runtime: RuneRuntime) {
        if (moduleInstance != null) {
            android.util.Log.w("RuneAsyncStorage", "Module already initialized")
            return
        }

        moduleInstance = RuneAsyncStorageModule(activity.applicationContext)
        runtime.installModules(listOf(moduleInstance!!))
        android.util.Log.d("RuneAsyncStorage", "Module initialized")
    }

    /**
     * Clean up the module (called during app teardown).
     */
    @JvmStatic
    fun cleanup() {
        moduleInstance = null
    }
}
